### ISSUE TITLE
Cann't change "linewidth" when adding geometries.

### ISSUE THREAD

**Issue by gepcel** [conda-forge/cartopy-feedstock#39]
There's a mismatch file between [cartopy of conda-forge](https://anaconda.org/conda-forge/cartopy/0.15.1/download/win-64/cartopy-0.15.1-py36_4.tar.bz2) and cartopy from master repository.

There is a Line 111 in file Lib/site-packages/cartopy/mpl/feature_artist.py :
```python
100        self.set_zorder(1)
111  self._kwargs.setdefault('linewidth', 1.0)
113  self._feature = feature
```
Which caused an issue that when `ax.add_geometries(..., linewidth=0.5)`, the 'linewidth' arg doesn't work. An example code might be like:
```python
%matplotlib inline
import matplotlib.pyplot as plt
import cartopy.crs as ccrs
from cartopy.io.shapereader import Reader
import cartopy.io.shapereader as shpreader

shpfilename = shpreader.natural_earth(resolution='110m',
                                      category='cultural',
                                      name='admin_0_countries')
shp = Reader(shpfilename)
fig, [ax1, ax2] = plt.subplots(1, 2, figsize=(8,3),
                               subplot_kw={'projection': ccrs.PlateCarree()})
ax1.add_geometries(shp.geometries(), crs=ccrs.PlateCarree(),
                 linewidth=0.1, edgecolor='k')
ax1.set(title='linewidth = 0.1')
ax2.add_geometries(shp.geometries(), crs=ccrs.PlateCarree(),
                 linewidth=10, edgecolor='k')
ax2.set(title='linewidth = 10')
plt.tight_layout()
plt.show()
```
The result would be:
![](https://user-images.githubusercontent.com/4179106/31042775-4cd2457a-a5e2-11e7-9c2b-f7724b46ef87.png)

As the line of code, I don't quite understand, because it seems to only to set linewidth to 1.0 if linewidth is not specified. Not sure what's wrong. I've filed an [issue](https://github.com/SciTools/cartopy/issues/937) at cartopy.

os: windows 10 x64
python: 3.6.2
cartopy: 0.15.1
matplotlib: 2.0.2
conda: 4.3.27

**Comment by ocefpaf** [comment]
@dopplershift I believe that is related to the patch you ported, correct? Can you take a look?

**Comment by gepcel** [comment]
OK, I pulled a [request](https://github.com/conda-forge/cartopy-feedstock/pull/40). I filed an issue first mainly because:

1. I don't understand why that line of code will make the `linewidth` unchangable. I suppose that will only make the linewidth default to 1.0, if no arg given.
2. Since there's no this line in the cartopy master repository, I'm not sure if there are other reasons of adding that line. So I want to ask first.
